sylt-compiler/src/lua.rs

```diff
@@ -31,16 +29,15 @@ macro_rules! write {
 pub struct LuaCompiler<'t> {
     compiler: &'t mut Compiler,
     loops: Vec<usize>,
-    pub file: File,
+    pub file: Box<dyn Write>,
 }
 
 impl<'t> LuaCompiler<'t> {
-    pub(crate) fn new(compiler: &'t mut Compiler, file: &Path) -> Self {
-        let file = File::create(&file).unwrap();
+    pub(crate) fn new(compiler: &'t mut Compiler, file: Box<dyn Write>) -> Self {
         Self {
             compiler,
             loops: Vec::new(),
-            file,
+            file: Box::new(file),
```

Comment:
```suggestion
            file,
```
I tested and it does not need two boxes.